### ISSUE TITLE
parameter count of Llama2-70B and Llama2-13B

### ISSUE THREAD

**Issue by joyjitkundu032** [meta-llama/llama#1111]
Hi All,

 I am struggling to get a count of 70B parameters for Llama2-70B model. Here is my calculation:
--------------------------------
Attention parameters per layer: 4 x 8192 x 8192
MLP parameters per layer (gate, up and down projection): 3 x 8192 x 28672
80 layers, vocab size 32000 (embedding dim 8192)

Total parameters ~ 80 x (4 x 8192 x 8192 + 3 x 8192 x 28672) + 32000 x 8192 ~ **78B**
Where am I getting it wrong?
---------------------------------
I do get correct count for 13B:
Total parameters ~ 40 x (4 x 5120 x 5120 + 3 x 5120 x 13824) + 32000 x 5120 ~ **12.7B**

Is it because of **grouped query** for 70B model?

**Comment by konnase** [comment]
Same question, any replies?

**Comment by konnase** [comment]
You are right, Llama2-70B use grouped query, so the parameters of qkov are not `4h^2`, instead qo is still `2h^2`, but kv is `kv_heads/attn_heads*2h^2`. So the total parameters of attention are `2h^2+kv_heads/attn_heads*2h^2`

![image](https://github.com/user-attachments/assets/9da5f04b-331a-4347-bea4-a16ee0518eaf)

**Comment by ajtarraga** [comment]
And in Llama2-13B, the MLP layer, why is it _3 x 5120 x 13824_?

I understand that 5120 is the hidden layer and 3 is the number of inputs and outputs, but the 13824?

I want to calculate the number of FLOPS per layer, can you help me?

**Comment by konnase** [comment]
> And in Llama2-13B, the MLP layer, why is it _3 x 5120 x 13824_?
> 
> I understand that 5120 is the hidden layer and 3 is the number of inputs and outputs, but the 13824?
> 
> I want to calculate the number of FLOPS per layer, can you help me?

13824 is intermediate_size

![Image](https://github.com/user-attachments/assets/81b0b7a7-6da6-4f32-9a4e-84dd126abf0b)
